### PR TITLE
Retain queries and convert `::` paths on redirects to rust-lang.org

### LINKED ISSUE #751: docs.rs/std/<path>?<args> should redirect path and args to doc.rust-lang
it trips me up when docs.rs/std/ works but docs.rs/std/io doesn't

### PR DESCRIPTION
**Pull request by Nemo157**
fixes #751, but using `::` separated paths instead of the `/` ones mentioned in the issue.

### CODE REVIEW

File: src/web/rustdoc.rs [hunk 1]
```diff
@@ -155,6 +134,12 @@ pub fn rustdoc_redirector_handler(req: &mut Request) -> IronResult<Response> {
         Some((krate, path)) => (krate.to_string(), Some(path.to_string())),
         None => (crate_name.to_string(), None),
     };
+
+    if let Some(inner_path) = DOC_RUST_LANG_ORG_REDIRECTS.get(crate_name.as_str()) {
```
Comment (syphar) [inline src/web/rustdoc.rs:138]: it feels strange to put this into the rustdoc-handler, in theory we could refactor the shared behaviour into separate methods and then have separate handlers. 

But since axum forbids overlapping routes contrary to iron which just takes the first matching route, we would have to merge the handlers anyways to a certain extend. 

So this is OK with me.

**Review by syphar — APPROVED** [review]
- code looks good 
- tests look good 

Since the coverage on redirects is relatively good I'm also certain that nothing breaks :) 

Only edge-case I can think of is that if I add any query arg to this endpoint, the query won't be added: `https://docs.rs/std::path?some=arg` -> `https://doc.rust-lang.org/stable/std/?some=arg`. But IMO this doesn't matter



Manual test also works.
